test/test_stations.py

```diff
@@ -0,0 +1,239 @@
+                        },
+                    ]
+                ),
+            )
+
+        added_station_ids = station_user.add_stations_to_group(
+            station_group["groupId"],
+            json.dumps([{"name": "Station name", "lat": -43.62367659982135, "lng": 172.62646754804894}]),
+        )
+        print("Added stations", added_station_ids["stationIds"])
+        assert len(added_station_ids["stationIds"]) == 1
+
+        print("Make sure added stations are not too close to any others (we don't want overlapping stations)")
+        with pytest.raises(BadRequestError) as error:
+            station_user.add_stations_to_group(
+                station_group["groupId"],
+                json.dumps(
+                    [
+                        {"name": "Station name", "lat": -43.62367659982135, "lng": 172.62646754804894},
+                        {"name": "Too close station", "lat": -43.62367657911904, "lng": 172.62626365029485},
+                    ]
+                ),
+            )
+        print("Error", json.loads(str(error.value)))
+
+        print("Existing stations that are missing when uploading new stations should be set to 'retired'")
+        added_station_ids = station_user.add_stations_to_group(
+            station_group["groupId"],
+            json.dumps(
+                [
+                    {"name": "Station name", "lat": -43.62367659982135, "lng": 172.62646754804894},
+                    {"name": "Will be retired", "lat": -43.62367659982135, "lng": 172.8},
+                ]
+            ),
+        )
+        assert len(added_station_ids["stationIds"]) == 2
+
+        print("Existing stations that are missing when uploading new stations should be set to 'retired'")
+        added_station_ids = station_user.add_stations_to_group(
+            station_group["groupId"],
+            json.dumps([{"name": "Station name", "lat": -43.62367659982135, "lng": 172.62646754804894}]),
+        )
+        assert len(added_station_ids["stationIds"]) == 1
+        added_station_id = added_station_ids["stationIds"][0]
+        print("Added", added_station_ids, added_station_id)
+
+        # Get stations and make sure that "Will be retired" exists and is retired
+        current_stations = station_user.get_stations_for_group(station_group["groupId"])
+        retired = [x for x in current_stations["stations"] if x["name"] == "Will be retired"]
+        assert len(retired) == 1
+        assert retired[0]["retiredAt"] is not None
+
+        # Get stations and make sure that "Will be retired" exists and is retired (getting group by name)
+        current_stations = station_user.get_stations_for_group(station_group_name)
+        retired = [x for x in current_stations["stations"] if x["name"] == "Will be retired"]
+        assert len(retired) == 1
+        assert retired[0]["retiredAt"] is not None
+
+        print("Adding stations with updated lat/lng but same name should update the lat/lng")
```

Comment:
I'm not sure we should be changing locations rather than storing a new one.   But it isn't the simplest solution for now so it's ok.